libs/component-library/src/lib/table/row.tsx

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import {
+  Easing,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated';
+import { mixColor } from 'react-native-redash';
+import { FxBox, FxBoxProps, FxReanimatedBox } from '../box/box';
+import { FxFoldableContent } from '../foldable-content/foldableContent';
+import { FxChevronDownIcon } from '../icons/icons';
+import { FxSpacer } from '../spacer/spacer';
+import { useFxTheme } from '../theme/useFxTheme';
+
+type RowProps = Omit<FxBoxProps, 'children'> & {
+  children: React.ReactNode;
+  showSeparator?: boolean;
+};
+export const Row = ({ children, showSeparator = true, ...rest }: RowProps) => {
+  return (
+    <FxBox
+      flexDirection="row"
+      paddingHorizontal="12"
+      paddingVertical="16"
+      alignItems="center"
+      borderBottomWidth={showSeparator ? 1 : 0}
+      borderBottomColor="backgroundSecondary"
+      {...rest}
+    >
+      {children}
+    </FxBox>
+  );
+};
+
+const ANIMATION_DURATION = 300;
+
+type RowGroupProps = {
+  firstRow: React.ReactNode;
+  hiddenRow: React.ReactNode;
+  iconWidth?: number;
+  showSeparator?: boolean;
+};
+
+export const RowGroup = ({
+  firstRow,
+  hiddenRow,
+  iconWidth = 32,
+  showSeparator = true,
+}: RowGroupProps) => {
+  const rotation = useSharedValue(0);
+  const backgroundAnimationValue = useSharedValue(0);
+  const theme = useFxTheme();
+
+  const iconAnimatedStyles = useAnimatedStyle(() => {
+    return {
+      transform: [{ rotateZ: `${rotation.value}deg` }],
+    };
+  });
+
+  const backgroundAnimatedStyle = useAnimatedStyle(() => {
+    return {
+      backgroundColor: mixColor(
+        backgroundAnimationValue.value,
+        theme.colors.backgroundApp,
+        theme.colors.backgroundPrimary
+      ) as string,
```

Comment:
Should we update the type definition for the `style` prop type in `FxReanimatedBox` rather than have to overwrite the return type here.

Reply:
I'm not sure if i want to override the `style` prop just to accommodate this one usage. What do you have in mind?

Reply:
Yeah, I looked at the `style` prop for `Animated.View` and it seems they make the type `any`. Not sure that is a good solution but it does allow you to pass `AnimatedColor` with out type errors. Very happy to keep as is though.